src/pytest/test_if.py

```diff
@@ -0,0 +1,136 @@
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if !@txt||to_number("redis") FIELDS txt word').error()
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if @empty||to_number("redis") FIELDS txt word').error()
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if !@empty||to_number("redis") FIELDS txt word').equal('OK')
+
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if !@txt&&to_number("redis") FIELDS txt word').equal('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if @empty&&to_number("redis") FIELDS txt word').equal('NOADD')
+
+def testExists(env):
+    env.cmd('FT.CREATE idx SCHEMA txt TEXT num NUMERIC empty TEXT')
+    env.cmd('FT.ADD idx doc1 1.0 FIELDS txt word num 10')
+    env.expect('FT.GET idx doc1').equal(['txt', 'word', 'num', '10'])
+
+    # test single field
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if exists(@txt) FIELDS txt word').equal('OK')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if exists(@num) FIELDS txt word').equal('OK')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if exists(@empty) FIELDS txt word').equal('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if exists(@noexist) FIELDS txt word').contains('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if !exists(@txt) FIELDS txt word').equal('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if !exists(@num) FIELDS txt word').equal('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if !exists(@empty) FIELDS txt word').equal('OK')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if !exists(@noexist) FIELDS txt word').contains('OK')
+
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if exists(@txt)||exists(@txt) FIELDS txt word').equal('OK')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if exists(@txt)||exists(@empty) FIELDS txt word').equal('OK')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if exists(@empty)||exists(@txt) FIELDS txt word').equal('OK')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if exists(@empty)||exists(@empty) FIELDS txt word').equal('NOADD')
+
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if exists(@txt)&&exists(@txt) FIELDS txt word').equal('OK')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if exists(@txt)&&exists(@empty) FIELDS txt word').equal('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if exists(@empty)&&exists(@txt) FIELDS txt word').equal('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if exists(@empty)&&exists(@empty) FIELDS txt word').equal('NOADD')
+
+    # check no crash
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if lower(exists(@empty)) FIELDS txt word').equal('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if upper(exists(@empty)) FIELDS txt word').equal('NOADD')
+    env.expect('FT.ADD idx doc1 1.0 REPLACE PARTIAL if substr(exists(@empty)) FIELDS txt word').error()
```

Comment:
I guess this is an inconsistent behavior, why substr return error but upper not. Lets keep it as is for now, we need to open another bigger issue to arrange this all.